### PR TITLE
CPM-625: Use UpsertProductCommand in postAction of ProductController

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

- Add a QueryMessageBus to handle the convertion from standardFormat to UserIntents
- Use this messageBus in internal API's ProductController for postAction

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            $userIntents[] = new ChangeParent($data['parent']);
+        }
+
+        if (\array_key_exists('values', $data)) {
+            $codes = \array_keys($data['values']);
+            $attributes = $this->attributeRepository->getAttributeTypeByCodes($codes);
+
+            $dataValues = $data['values'];
+
+            foreach ($attributes as $attributeCode => $attributeType) {
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php:90]: foreach($data as $attributeCode => $values)

File: src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                            $locale,
+                            \array_map(fn ($measurement) => new PriceValue($measurement['amount'], $measurement['currency']), $attributeData)
+                        ),
+                        AttributeTypes::TEXTAREA => new SetTextareaValue($attributeCode, $scope, $locale, $attributeData),
+                        // TODO: use SetTableValue
+                        AttributeTypes::TEXT, AttributeTypes::TABLE => new SetTextValue($attributeCode, $scope, $locale, $attributeData),
+                        AttributeTypes::REFERENCE_ENTITY_SIMPLE_SELECT => new SetSimpleReferenceEntityValue($attributeCode, $scope, $locale, $attributeData),
+                        AttributeTypes::REFERENCE_ENTITY_COLLECTION => new SetMultiReferenceEntityValue($attributeCode, $scope, $locale, $attributeData),
+                        AttributeTypes::ASSET_COLLECTION => new SetAssetValue($attributeCode, $scope, $locale, $attributeData),
+                        default => throw new \LogicException('Attribute type has no user intent associated')
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php:122]: Attribute does not exists

File: src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+    /**
+     * @param array<string, mixed> $data
+     * @return UserIntent[]
+     */
+    private function getUserIntentsFromData(array $data): array
+    {
+        $userIntents = [];
+
+        if (\array_key_exists('family', $data) && null !== $data['family']) {
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php:68]: remove null !== $fata['family']

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public function create(string $attributeType, string $attributeCode, mixed $data): ValueUserIntent
+    {
+        $priceValues = [];
+        $this->validateValueStructure($attributeCode, $data);
+        if (null === $data['data'] || [] === $data['data']) {
+            return new ClearValue($attributeCode, $data['scope'], $data['locale']);
+        }
+        foreach ($data['data'] as $measurement) {
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php:36]: $measurement -> $price ?

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            $this->validateScalarArray($attributeCode, $measurement);
+            if (!array_key_exists('amount', $measurement)) {
+                throw InvalidPropertyTypeException::arrayKeyExpected($attributeCode, 'amount', static::class, $data);
+            }
+            if (!array_key_exists('currency', $measurement)) {
+                throw InvalidPropertyTypeException::arrayKeyExpected($attributeCode, 'currency', static::class, $data);
+            }
+            if (null === $measurement['amount'] || '' === $measurement['amount']) {
+                return new ClearValue($attributeCode, $data['scope'], $data['locale']);
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php:45]: If one of the prices has an empty amount, you clear the all attribute.
Should'nt you just ignore the current price and not add it to the $priceValues array ?

**Review by amigaire — COMMENTED** [review]
GTM after fix

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php [hunk 6]
```diff
@@ -165,32 +169,78 @@ public function createAction(Request $request)
...
-     * @param Request $request
-     * @param string  $id
-     *
      * @throws NotFoundHttpException     If product is not found or the user cannot see it
      * @throws AccessDeniedHttpException If the user does not have right to edit the product
-     *
-     * @return Response
      */
-    public function postAction(Request $request, $id)
+    public function postAction(Request $request, string $id): Response
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php:175]: To move in its own controller?

File: src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * @return UserIntent[]
+     */
+    public function __invoke(GetUserIntentsFromStandardFormat $getUserIntentsFromStandardFormat): array
+    {
+        $standardFormat = $getUserIntentsFromStandardFormat->standardFormat();
+
+        $userIntents = [];
+        foreach ($standardFormat as $fieldName => $data) {
+            $result = $this->userIntentFactoryRegistry->fromStandardFormatField($fieldName, $data);
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Application/StandardFormat/ConvertStandardFormatIntoUserIntentsHandler.php:33]: Why not always return an array (for instance: [SetEnabled(true)])

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/UserIntentFactoryRegistry.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }
+    }
+
+    /**
+     * @param string $fieldName
+     * @param mixed $data
+     * @return UserIntent|UserIntent[]|null
+     */
+    public function fromStandardFormatField(string $fieldName, mixed $data): UserIntent | array | null
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/UserIntentFactoryRegistry.php:44]: return `array | null`

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/CategoriesUserIntentFactory.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    use ValidateDataTrait;
+
+    public function getSupportedFieldNames(): array
+    {
+        return ['categories'];
+    }
+
+    public function create(string $fieldName, mixed $data): UserIntent|array
+    {
+        if (null === $data) {
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/CategoriesUserIntentFactory.php:26]: Check if we really want to remove all categories when user sends null

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/GroupsUserIntentFactory.php [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    use ValidateDataTrait;
+
+    public function getSupportedFieldNames(): array
+    {
+        return ['groups'];
+    }
+
+    public function create(string $fieldName, mixed $data): UserIntent|array
+    {
+        if (null === $data) {
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/GroupsUserIntentFactory.php:26]: check if null means clear value

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php [hunk 11]
```diff
@@ -414,6 +362,66 @@ protected function updateProduct(ProductInterface $product, array $data)
             $data = $this->productAttributeFilter->filter($data);
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php:362]: remove updateProduct

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/UserIntentFactoryRegistry.php [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $factory = $this->userIntentFactoriesWithFieldName[$fieldName] ?? null;
+        if (null === $factory && \in_array($fieldName, $this->ignoredFieldNames)) {
+            return null;
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/UserIntentFactoryRegistry.php:48]: return [];

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/IdentifierValueUserIntentFactory.php [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+use Akeneo\Pim\Enrichment\Product\API\Command\UserIntent\ValueUserIntent;
+use Akeneo\Pim\Enrichment\Product\Domain\UserIntent\Factory\ValueUserIntentFactory;
+use Akeneo\Pim\Structure\Component\AttributeTypes;
+use Akeneo\Tool\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IdentifierValueUserIntentFactory implements ValueUserIntentFactory
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/IdentifierValueUserIntentFactory.php:18]: remove (identifier is ignored)

File: src/Akeneo/Pim/Enrichment/Product/back/Infrastructure/Symfony/Resources/config/factories.yml [hunk 14]
```diff
@@ -0,0 +1,74 @@
+services:
+    Akeneo\Pim\Enrichment\Product\Domain\UserIntent\UserIntentFactoryRegistry:
+        arguments:
+            - !tagged_iterator 'pim.enrichment.product.user_intent_factory'
+            - ['identifier', 'created', 'updated', 'parent_associations', 'metadata', 'quality_scores', 'completenesses']
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Infrastructure/Symfony/Resources/config/factories.yml:5]: remove 'identifier'

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/DateValueUserIntentFactory.php [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+use Akeneo\Tool\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class DateValueUserIntentFactory implements ValueUserIntentFactory
+{
+    use ValidateDataTrait;
+    private const PATTERN = '/^(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})(T(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)(([-+](\d{2}):(\d{2})|Z)?))?$/';
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/DateValueUserIntentFactory.php:23]: check if format changes according to locale

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MeasurementValueUserIntentFactory.php [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        if (null === $data['data']['amount'] || '' === $data['data']['amount']) {
+            return new ClearValue($attributeCode, $data['scope'], $data['locale']);
+        }
+
+        return new SetMeasurementValue(
+            $attributeCode,
+            $data['scope'],
+            $data['locale'],
+            $data['data']['amount'],
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MeasurementValueUserIntentFactory.php:49]: check amount value type before

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+class MultiStringValueUserIntentFactory implements ValueUserIntentFactory
+{
+    use ValidateDataTrait;
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::OPTION_MULTI_SELECT,
+            AttributeTypes::REFERENCE_ENTITY_COLLECTION,
+            AttributeTypes::REFERENCE_DATA_MULTI_SELECT,
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php:29]: missing userIntents for REFERENCE_DATA_MULTI_SELECT and REFERENCE_DATA_SIMPLE_SELECT

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        $this->validateValueStructure($attributeCode, $data);
+        if (null === $data['data']) {
+            return new ClearValue($attributeCode, $data['scope'], $data['locale']);
+        }
+        $this->validateScalarArray($attributeCode, $data['data']);
+
+        return match ($attributeType) {
+            AttributeTypes::OPTION_MULTI_SELECT => new SetMultiSelectValue($attributeCode, $data['scope'], $data['locale'], $data['data']),
+            AttributeTypes::REFERENCE_ENTITY_COLLECTION, AttributeTypes::REFERENCE_DATA_MULTI_SELECT => new SetMultiReferenceEntityValue($attributeCode, $data['scope'], $data['locale'], $data['data']),
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php:44]: use correct userintent AttributeTypes::REFERENCE_DATA_MULTI_SELECT

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+            if (!array_key_exists('amount', $price)) {
+                throw InvalidPropertyTypeException::arrayKeyExpected($attributeCode, 'amount', static::class, $data);
+            }
+            if (!array_key_exists('currency', $price)) {
+                throw InvalidPropertyTypeException::arrayKeyExpected($attributeCode, 'currency', static::class, $data);
+            }
+            if (null === $price['amount'] || '' === $price['amount']) {
+                continue;
+            }
+            $priceValues[] = new PriceValue($price['amount'], $price['currency']);
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/PriceCollectionValueUserIntentFactory.php:47]: check amount and currency types

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::TEXT,
+            AttributeTypes::TEXTAREA,
+            AttributeTypes::FILE,
+            AttributeTypes::OPTION_SIMPLE_SELECT,
+            AttributeTypes::IMAGE,
+            AttributeTypes::REFERENCE_DATA_SIMPLE_SELECT,
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php:37]: remove for now

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+class StringValueUserIntentFactory implements ValueUserIntentFactory
+{
+    use ValidateDataTrait;
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::TEXT,
+            AttributeTypes::TEXTAREA,
+            AttributeTypes::FILE,
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php:34]: Move File and Image in FileValueUserIntentFactory

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/TableValueUserIntentFactory.php [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+    public function getSupportedAttributeTypes(): array
+    {
+        return [AttributeTypes::TABLE];
+    }
+
+    public function create(string $attributeType, string $attributeCode, mixed $data): ValueUserIntent
+    {
+        $this->validateValueStructure($attributeCode, $data);
+
+        return new SetTableValue($attributeCode, $data['scope'], $data['locale'], $data['data']);
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/TableValueUserIntentFactory.php:30]: check data['data'] is array

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/ValueUserIntentFactoryRegistry.php [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * @inerhitDoc
+     */
+    public function create(string $fieldName, mixed $data): UserIntent | array
+    {
+        Assert::isArray($data);
+
+        $attributeTypesByCode = $this->attributeRepository->getAttributeTypeByCodes(\array_keys($data));
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/ValueUserIntentFactoryRegistry.php:51]: make it case insensitive (array_change_key_case)

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/ValueUserIntentFactoryRegistry.php [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+        $valueUserIntents = [];
+        foreach ($data as $attributeCode => $values) {
+            $attributeType = $attributeTypesByCode[$attributeCode] ?? null;
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/ValueUserIntentFactoryRegistry.php:54]: strtolower($attributeCode)

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/DateValueUserIntentFactory.php (hunk 15 above)
Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/DateValueUserIntentFactory.php:23]: The localized converter (AttributeConverterInterface) converts the localized date into ISO8601 format. So the pattern is correct

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/Product/UpdateProductController.php [hunk 25]
```diff
@@ -0,0 +1,295 @@
...
+    /**
+     * Find a product by its id or return a 404 response
+     *
+     * @param string $id the product id
+     *
+     * @throws NotFoundHttpException
+     *
+     * @return ProductInterface
+     */
+    private function findProductOr404($id)
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/Product/UpdateProductController.php:143]: As feature branch is merged, should this not be a $uuid instead ?
If yes, change the variable names and the message above

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/FileValueUserIntentFactory.php [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+            return new ClearValue($attributeCode, $data['scope'], $data['locale']);
+        }
+        if (!\is_string($data['data'])) {
+            throw InvalidPropertyTypeException::stringExpected($attributeCode, static::class, $data['data']);
+        }
+
+        return match ($attributeType) {
+            AttributeTypes::FILE => new SetFileValue($attributeCode, $data['scope'], $data['locale'], $data['data']),
+            AttributeTypes::IMAGE => new SetImageValue($attributeCode, $data['scope'], $data['locale'], $data['data']),
+            default => throw new \InvalidArgumentException('Not implemented')
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/FileValueUserIntentFactory.php:45]: Why 1 file for the 2 attribute types ? This should prevent this weird default use case.

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+class MultiStringValueUserIntentFactory implements ValueUserIntentFactory
+{
+    use ValidateDataTrait;
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::OPTION_MULTI_SELECT,
+            AttributeTypes::REFERENCE_ENTITY_COLLECTION,
+            // TODO: Add when userIntent is ready
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php:29]: Please add ticket number

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+{
+    use ValidateDataTrait;
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::TEXT,
+            AttributeTypes::TEXTAREA,
+            AttributeTypes::OPTION_SIMPLE_SELECT,
+            // TODO: Add when userIntent is ready
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/StringValueUserIntentFactory.php:32]: Same

**Review by pierallard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php (hunk 27 above)
Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php]: Done. Now we have to do this ticket before merging this PR

File: src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+class MultiStringValueUserIntentFactory implements ValueUserIntentFactory
+{
+    use ValidateDataTrait;
+
+    public function getSupportedAttributeTypes(): array
+    {
+        return [
+            AttributeTypes::OPTION_MULTI_SELECT,
+            AttributeTypes::REFERENCE_ENTITY_COLLECTION,
+            // TODO CPM-673: Add when userIntent is ready
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Domain/UserIntent/Factory/Value/MultiStringValueUserIntentFactory.php:29]: TODO

**Pull request merged**